subsys/net/openthread/rpc/server/ot_rpc_ctrl_server.c

```diff
@@ -116,6 +117,35 @@ static void ot_rpc_cmd_ip6_get_unicast_addrs(const struct nrf_rpc_group *group,
 	nrf_rpc_cbor_rsp_no_err(group, &rsp_ctx);
 }
 
+static void ot_rpc_cmd_ip6_get_multicast_addrs(const struct nrf_rpc_group *group,
+					       struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	struct nrf_rpc_cbor_ctx rsp_ctx;
+	const otNetifMulticastAddress *addr;
+	size_t addr_count;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
```

Comment:
Is it mandatory for any open thread api call?

Reply:
Yes, per nRF RPC architecture there can be multiple handling threads so we need a synchronization at OT level.